common/src/main/java/org/fao/geonet/utils/Xml.java

```diff
@@ -485,6 +495,13 @@ protected static Path resolvePath(Source s) throws URISyntaxException {
                     for (Map.Entry<String, Object> param : params.entrySet()) {
                         t.setParameter(param.getKey(), param.getValue());
                     }
+
+                if (params.containsKey("geonet-force-xml")) {
+                    ((Controller) t).setOutputProperty("indent", "yes");
+                    ((Controller) t).setOutputProperty("method", "xml");
+                    ((Controller) t).setOutputProperty("{http://saxon.sf.net/}indent-spaces", "3");
```

Comment:
is "4" not a more usual indent factor ? ;-)

Reply:
2 is even the value used in GN source code for XML https://github.com/geonetwork/core-geonetwork/blob/master/.editorconfig#L3

As ISO19139 can be quite nested, I would opt for a small indent size.